libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushMessaging.kt

```diff
@@ -473,12 +474,14 @@ object PushMessaging {
             enqueuePushNotificationsRegistrationWork(
                 userAccount = null,
                 action = action,
+                pushNotificationsRegistrationType = ReRegistrationOnAppForeground,
```

Comment:
I did find one logic fix to make from the previous pull request.  Regardless of the app's chosen push notification re-registration type, push notification registration has to happen once when the token is obtained.  So, I added an override in this call to allow that.